test/functional/wallet_backwards_compatibility.py

```diff
@@ -378,6 +394,23 @@ def run_test(self):
 
             wallet.unloadwallet()
 
+            # Open the wallet with sqlite and inspect the flags and records
+            if SQLITE3_IMPORTED:
+                conn = sqlite3.connect(down_backup_path)
+                with conn:
+                    flags_rec = conn.execute("SELECT value FROM main WHERE key = x'05666c616773'").fetchone()
+                    new_flags = int.from_bytes(flags_rec[0], byteorder="little")
+                    diff_flags = new_flags & ~old_flags
+
+                    # Check for last hardened xpubs if the flag is newly set
+                    if diff_flags & (1 << 2):
+                        self.log.debug("Checking descriptor cache was upgraded")
+                        # Fetch all records with the walletdescriptorlhcache prefix
+                        lh_cache_recs = conn.execute("SELECT value FROM main where key >= x'1777616c6c657464657363726970746f726c686361636865' AND key < x'1777616c6c657464657363726970746f726c686361636866'").fetchall()
```

Comment:
In 3854d6b56c50b26f7b61dfb37f5230726d561cf9 "tests: Check that the last hardened cache upgrade occurs"

At the stage where there are multiple such instances of hardcoding the hex representation of the database keys in the tests, I would introduce a `get_db_key_hex` utility. Mentioned in overall review comment.

Reply:
I've changed this to use the existing `ser_string` utility which is more accurate.

Reply:
I see, yes `ser_string` seems thorough.